contracts/mainnet/connectors/basic-ERC4626/main.sol

```diff
@@ -17,57 +17,61 @@ abstract contract BasicConnector is Events, DSMath, Basic {
 	/**
 	 * @dev Deposit underlying asset to ERC4626 Vault.
 	 * @notice Mints vault shares by depositing exactly amount of underlying assets
-	 * @param token ERC4626 Token address.
+	 * @param vaultToken ERC4626 Token address.
 	 * @param underlyingAmt The amount of the underlying asset to deposit. (For max: `uint256(-1)`)
-	 * @param minSharesPerToken The min share rate of deposit
+	 * @param minSharesPerToken The min share rate of deposit. Should always be in 18 decimals.
 	 * @param getId ID to retrieve amt.
 	 * @param setId ID stores the amount of tokens deposited.
 	 */
-
 	function deposit(
-		address token,
+		address vaultToken,
 		uint256 underlyingAmt,
 		uint256 minSharesPerToken,
 		uint256 getId,
 		uint256 setId
 	) public returns (string memory _eventName, bytes memory _eventParam) {
 		uint256 _underlyingAmt = getUint(getId, underlyingAmt);
-		IERC4626 vaultTokenContract = IERC4626(token);
 
-		address _underlyingToken = vaultTokenContract.asset();
-		uint8 _vaultShareDecimal = vaultTokenContract.decimals();
-		TokenInterface underlyingTokenContract = TokenInterface(
-			_underlyingToken
+		IERC4626 vaultTokenContract = IERC4626(vaultToken);
+		TokenInterface _underlyingTokenContract = TokenInterface(
+			vaultTokenContract.asset()
 		);
 
 		_underlyingAmt = _underlyingAmt == uint256(-1)
-			? underlyingTokenContract.balanceOf(address(this))
+			? _underlyingTokenContract.balanceOf(address(this))
 			: _underlyingAmt;
 
+		// Making sure that the underlying amount is always in 18 decimals.
+		uint256 _underlyingAmt18 = convertTo18(
+			_underlyingTokenContract.decimals(),
+			_underlyingAmt
+		);
+
 		uint256 _minShares = convert18ToDec(
-			_vaultShareDecimal,
-			wmul(minSharesPerToken, _underlyingAmt)
+			vaultTokenContract.decimals(),
+			wmul(minSharesPerToken, _underlyingAmt18) // Returns final amount in the floor value of the two token decimals.

```

Comment:
We can simplify this whole codebase by removing `convertTo18` & `convert18ToDec`. We should do that.

Reply:
Updated in https://github.com/Instadapp/dsa-connectors/pull/310/commits/2424035fa491df2194dcb5f0635bedc985d486b9